app/src/main/java/com/skyyo/samples/features/videopager/composables/VideoPager.kt

```diff
@@ -0,0 +1,113 @@
+package com.skyyo.samples.features.videopager.composables
+
+import androidx.compose.foundation.gestures.Orientation
+import androidx.compose.foundation.gestures.draggable
+import androidx.compose.foundation.gestures.rememberDraggableState
+import androidx.compose.foundation.layout.Column
+import androidx.compose.runtime.*
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.layout.Layout
+import androidx.compose.ui.layout.Measurable
+import androidx.compose.ui.layout.ParentDataModifier
+import androidx.compose.ui.unit.Density
+import com.skyyo.samples.features.videopager.model.PagerState
+import kotlinx.coroutines.launch
+import kotlin.math.roundToInt
+
+@Immutable
+private data class PageData(val page: Int) : ParentDataModifier {
+    override fun Density.modifyParentData(parentData: Any?): Any = this@PageData
+}
+
+data class PagerScope(val state: PagerState, val page: Int)
+
+@Composable
+fun VideoPager(
```

Comment:
might be a good idea to look into `pager` itself. It might have what we need for this. 
Also we must consider infinite amount of items, so that we need some sort of LazyList. 
That's where the pager can help, since it's based on LazyColumn/Row